receiver/prometheusreceiver/metrics_receiver_open_metrics_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusreceiver
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+const testDir = "./testdata/openmetrics/"
+
+func verifyPositiveTarget(t *testing.T, _ *testData, mds []*pdata.ResourceMetrics) {
+	require.Greater(t, len(mds), 0, "At least one resource metric should be present")
+	metrics := getMetrics(mds[0])
+	assertUp(t, 1, metrics)
+}
+
+// Test open metrics positive test cases
+func TestOpenMetricsPositive(t *testing.T) {
+	targetsMap := getOpenMetricsTestData(false)
+	targets := make([]*testData, 0)
+	for k, v := range targetsMap {
+		testData := &testData{
+			name: k,
+			pages: []mockPrometheusResponse{
+				{code: 200, data: v},
+			},
+			validateFunc: verifyPositiveTarget,
+		}
+		targets = append(targets, testData)
+	}
+
+	testComponent(t, targets, false, "", true)
+}
+
+func verifyNegativeTarget(t *testing.T, _ *testData, mds []*pdata.ResourceMetrics) {
+	// negative tests are skipped since prometheus scrape package is currently not fully
```

Comment:
Are some of the tests supported?  It would be nice to run the tests that work, and only skip the tests that don't work.

Maybe add a deny-list for files containing tests we don't support at the top?

Reply:
Thanks for the review. Yes, some of the tests are supported. Pushed a change to only skip tests that aren't supported.